### PR TITLE
Add support for assignments to and from large (>64-bit) operand

### PR DESCRIPTION
**Pull request by usha1830**
To support assignments between variables <= 64-bit and >64-bit variables, the smaller operand must be a header field.

For ex:
```
 bit<128> var1;
bit<32> var2;
var 1 = (bit<128>) var2  

```
var2 should be moved to a  header to make this assignment work on the DPDK target.
Compiler creates a pseudo header for holding the variables that need to be a header field and inserts additional assignments to move the smaller operand in a pseudo header field

```
header pseudo_hdr {
bit<32> pseudo;
}
pseudo = var2
var1 = (bit<128>)pseudo;
```

### CODE REVIEW

File: backends/dpdk/dpdkUtils.cpp [hunk 1]
```diff
@@ -75,6 +75,24 @@ bool isEightBitAligned(const IR::Expression *e) {
     return true;
 }
 
+bool isLargeFieldOperand(const IR::Expression *e) {
+    auto expr = e;
+    if (auto base = e->to<IR::Cast>()) expr = base->expr;
+    if (auto type = expr->type->to<IR::Type_Bits>()) {
```
Comment (psivanup) [inline backends/dpdk/dpdkUtils.cpp:81]: is slice expression expected here?

**Review by psivanup — APPROVED** [review]
Looks fine.

File: backends/dpdk/dpdkUtils.cpp (hunk 1 above)
Comment (usha1830) [reply backends/dpdk/dpdkUtils.cpp:81]: Yes. We may get slice, but currently there is no support for  large operands for other operations such as shift and bitwise AND. Hence, we are not making any changes w.r.t that. Only simple assignments and casts are considered for now.

**Comment by usha1830** [comment]
Closing as upstreamed

**Pull request closed without merging**